lib/pure/strutils.nim

```diff
@@ -154,6 +154,122 @@ proc isDigit*(c: char): bool {.noSideEffect,
     doAssert isDigit('8') == true
   return c in Digits
 
+proc isNumeric*(s: string, enabledNaN = false): bool =
+  ## Checks whether the string is numeric.
+  ## When the string is an integer, float or exponential, it returns true,
+  ## otherwise it returns false.
+  ## 'enabledNaN' variable value indicates whether 'NaN', 'Inf', '.9 ', '9.'
```

Comment:
`enabledNaN` is a bad name for that; maybe: 
enableSpecial, enableLeadingDot, enableTrailingDot ?

Reply:
I vote for `strict`
https://github.com/nim-lang/Nim/pull/15903#discussion_r521112176

Reply:
I'd rather have a minimum of flexibility than 1 option fits all (in practice this means 1-fits-none), so it can be used in more contexts eg:
* supports whatever json considers as numbers (see https://restfulapi.net/json-data-types/)
* supports whatever nim considers numbers
* accept vs support loose wrt preceding 0 (eg: `010`, which json rejects)
* accept vs support loose wrt `dot` (eg: `5.`, and `.5`)

and the simplest is either a `set[Options]` or a set of bool flags like `enableNaNInf, enableLooseDot, enableLooseZero` or other partitioning